### ISSUE TITLE
c:geo logs wrong cache

### ISSUE THREAD

**Issue by Lineflyer** [cgeo/cgeo#6333]
I discussed this case on support mail already some moths ago. Now the user contacted me again with some details, and said he had at least 10 cases of this problem on his cache and all confirmed to be c:geo users.

Here is what he writes:
```
Wenn jemand über c:geo den Cache http://coord.info/GC41JNJ loggt,
erfolgt der Log nicht dort, sondern in meinem Cache
https://coord.info/GC15X39 Es ist unabhängig von den Versionen.
```
Short translation:
If a c:geo user composes a log for GC41JNJ the log will be posted to GC15X39 instead.

I contacted the owner of the other cache and asked him whether I can do a test log on his cache.
Will try to reproduce it.

If this is really a bug, what could be the cause? Some mixup with the GUID?

**Comment by Lineflyer** [comment]
GUID:
GC41JNJ   3b307f03-7949-4558-9581-86a15d50a0ee
GC15X39  a49a120e-998e-4a54-ba57-a0f4f24e3159

**Comment by pstorch** [comment]
Log IDs:
https://www.geocaching.com/seek/log.aspx?ID=3330722&lcn=1
https://www.geocaching.com/seek/log.aspx?ID=688366&lcn=1

**Comment by Lineflyer** [comment]
Kind of confirmed:
If I load GC41JNJ in c:geo and start the logging activity I do see:
```
22:18:09.396 Debug cgeo 8786  [OkHttp] GET https://www.geocaching.com/seek/log.aspx?ID=688366
```
This is the log page of the other cache GC15X39

**Comment by Lineflyer** [comment]
The cache GC41JNJ contains a hyperlink to the log page of GC15X39 in its description.
We accidentally parse this URL to be used for logging:
```
„Weil man sich dabei den Kopf zerbricht." </span> <a href="http://www.geocaching.com/seek/log.aspx?ID=688366#"```

Whyever he did it. Can we make the parsing match stricter to the correct URL?

**Comment by pstorch** [comment]
We could change the pattern to include the `id` attribute like this:
```
    static final Pattern PATTERN_CACHEID = Pattern.compile("/seek/log\\.aspx\\?ID=(\\d+)[^\"]*\"\\s*id=\"ctl00_ContentBody_GeoNav_logButton\"");
```

Then we have the correct match, but I'm thinking why we parse the CacheId from the HTML at all? 
We could also derive it from the geocode (via `GCConstants.gccodeToGCId(cache.getGeocode())`). That would be the best. Maybe there is something I don't see at the moment.

**Comment by Lineflyer** [comment]
I have built a new NB1 and did a quick test:
I refreshed GC41JNJ and started the logging activity and do still see a requests going out to https://www.geocaching.com/seek/log.aspx?ID=688366

Should the ID be overwritten by refresh of the cache details? Seems either this is not the case or the pattern is not matching.

**Comment by pstorch** [comment]
Maybe the old id is still in the database and not overwritten with null. @UniQP maybe the setCacheId() should be called.

**Comment by UniQP** [comment]
@Lineflyer Can you test the new PR?

**Comment by Lineflyer** [comment]
Assigned to myself for testing

**Comment by samueltardieu** [comment]
I'll add the Do not merge label to the PR in the meantime until you can test it.

**Comment by Lineflyer** [comment]
I thought it was already tested. Be aware I can only test changes already merged here, as I dont have the ability to build myself.

**Comment by pstorch** [comment]
Sorry, I've tested the first fix with a fresh DB. And the cache id was correct. Missed the point that a wrong id might not be updated. I can test later this evening.

**Comment by pstorch** [comment]
Looks good now. See my comment in PR.

**Comment by Lineflyer** [comment]
Confirmed working with the latest patch (tested in 2017.02.23-NB1-7223226)
Refreshed GC41JNJ and started logging. I do see the correct log ID now in the debug log.
